enclave/core/sgx/hostcalls.c

```diff
@@ -13,10 +13,14 @@
 static oe_ecall_context_t* _get_ecall_context()
 {
     oe_sgx_td_t* td = oe_sgx_get_td();
-    oe_ecall_context_t* ecall_context = td->host_ecall_context;
-    return oe_is_outside_enclave(ecall_context, sizeof(*ecall_context))
-               ? ecall_context
-               : NULL;
+    if (td)
+    {
+        oe_ecall_context_t* ecall_context = td->host_ecall_context;
+        return oe_is_outside_enclave(ecall_context, sizeof(*ecall_context))
```

Comment:
Add check to see if ecall_context is null as well.

Reply:
Should not be needed, oe_is_outside_enclave() already explicitly returns false if you pass in a null pointer